dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/RateLimiter.java

```diff
@@ -0,0 +1,224 @@
+            try {
+                rlEvent = new RateLimitEvent().tokenizeAndGetEvent(rateLimitPRCreation);
+            } catch (UnknownFormatConversionException ex) {
+                log.error("Failed to parse ratelimiting argument, will not impose any rate limits", ex);
+                return null;
+            }
+            log.info("Use rateLimiting is enabled, the PRs will be throttled in this run..");
+            return new RateLimiter(rlEvent.rateLimit, rlEvent.rateLimitDuration, rlEvent.tokenAddingRate);
+        }
+        log.info("Use rateLimiting is disabled, the PRs will not be throttled in this run..");
+        return null;
+    }
+
+    /**
+     * This method consumes a token from the bucked.
+     * It blocks the execution when no token is available
+     * for consumption.
+     *
+     * @throws InterruptedException when interrupted
+     * @see RateLimiter#bucket
+     */
+    public void consume() throws InterruptedException {
+        bucket.asBlocking().consume(1);
+    }
+
+    /**
+     * Creating this as inner class as existence of this class outside
+     * RateLimiter class will make a little sense. This class is solely
+     * meant to encapsulate the primitives needed for the RateLimiter
+     * and logic for populating those after tokenizing the input
+     */
+    class RateLimitEvent {
+        private final Pattern eventPattern =
+                Pattern.compile("(^[1-9]\\d{0,18})(-per-)?([1-9]\\d{0,18})*([smh]?)", Pattern.CASE_INSENSITIVE);
+        private final long rateLimit;
+        private final Duration rateLimitDuration;
+        private final Duration tokenAddingRate;
+
+        public RateLimitEvent(long rateLimit, Duration rateLimitDuration, Duration tokenAddingRate) {
+            this.rateLimit = rateLimit;
+            this.rateLimitDuration = rateLimitDuration;
+            this.tokenAddingRate = tokenAddingRate;
+        }
+
+        public RateLimitEvent() {
+            this(0, null, null);
+        }
+
+        public long getRateLimit() {
+            return rateLimit;
+        }
+
+        public Duration getRateLimitDuration() {
+            return rateLimitDuration;
+        }
+
+        public Duration getTokenAddingRate() {
+            return tokenAddingRate;
+        }
+
+        public RateLimitEvent tokenizeAndGetEvent(String input) {
+
+            if (input == null) {
+                throw new UnknownFormatConversionException("Can not tokenize/parse a null object");
+            }
+
+            long rLimit;
+            long duration;
+            Duration rLimitDuration;
+            Duration tokAddingRate;
+            String literalConstant;
+            String rateDuration;
+            String rateDurationUnitChar;
+
+            Matcher matcher = eventPattern.matcher(input);
+
+            if (!matcher.matches()) {
+                throw new UnknownFormatConversionException("input argument is not in valid format and can't be tokenize");
```

Comment:
Same suggestion: https://github.com/salesforce/dockerfile-image-update/pull/717/files#r1059085290

Reply:
You also have duplicate text - would be good to have the same error msg returned